### PR TITLE
feat: Add Webhook Hostname (#668)

### PR DESCRIPTION
**Pull request by arcdigital**

### CODE REVIEW

File: internal/daemon/daemon.go [hunk 1]
```diff
@@ -453,8 +456,13 @@ func (d *Daemon) Start(ctx context.Context, started chan struct{}) error {
 	if d.Host == "" {
 		listenAddress := ln.Addr().(*net.TCPAddr)
 		d.System.SetHostname(internal.NormalizeAddress(listenAddress))
+		if d.WebhookHost == "" {
+			d.System.SetWebhookHostname(internal.NormalizeAddress(listenAddress))
+		}
 	}
+
 	d.V(0).Info("set system hostname", "hostname", d.System.Hostname())
+	d.V(0).Info("set webhook hostname", "webhookHostname", d.System.WebhookHostname())
```
Comment (leg100) [inline internal/daemon/daemon.go:465]: Nitpick: I would refer not to `"webhookHostname`" but just to `"hostname"`, as the fact it is the hostname for the webhook is made clear in the log message itself, `"set webhook hostname"`.

File: internal/daemon/daemon.go [hunk 2]
```diff
@@ -93,6 +93,9 @@ func New(ctx context.Context, logger logr.Logger, cfg Config) (*Daemon, error) {
 	}
 
 	hostnameService := internal.NewHostnameService(cfg.Host)
+	if cfg.WebhookHost != "" {
+		hostnameService.SetWebhookHostname(cfg.WebhookHost)
+	}
```
Comment (leg100) [inline internal/daemon/daemon.go:98]: You might want to take this as an opportunity to refactor the hostname service. Currently, there is hostname "domain logic" here in `daemon.go` that really ought to belong in `hostname.go` instead. The logic I'm talking about is "if this is set, set this, if this is not set, then set that, etc.", which daemon should not care about, but the hostname service _should_ care about.

So have `daemon.go` just set values via the hostname service `Set(...)` methods. The hostname service then determines what to with them, or it's "getter" methods determine which values to return depending on what has been set.

I would do the same with the default hostname, `localhost:8080` - move that into `hostname.go`.

This way, as much as possible on the subject of hostnames is in `hostname.go`.

File: internal/daemon/daemon.go (hunk 1 above)
Comment (arcdigital) [reply internal/daemon/daemon.go:465]: I did it like that, so it is easier to parse out of the logs if somebody wanted to

File: internal/daemon/daemon.go (hunk 2 above)
Comment (arcdigital) [reply internal/daemon/daemon.go]: I removed this and moved that logic into `hostname.go`, I think the rest of the refactor could be handled in another PR so this change isn't blocked by that?

**Comment by leg100** [comment]
@arcdigital Looks good. I've made a few updates:

* Added unit tests that test the behaviour you reported to me in Slack
* Set the webhook hostname for Github Apps as well (they're managed separately from per-repo webhooks)
* Changed the logging key from `webhookHostname` to `webhook_hostname` (keeps it consistent with other multi word logging keys).

**Pull request merged**